src/include/baselib/jni/JavaVirtualMachine.h

```diff
@@ -157,69 +206,114 @@ namespace bl
                 }
             }
 
-            void createJvm()
+            ~JavaVirtualMachineT() NOEXCEPT
             {
-                BL_MUTEX_GUARD( g_lock );
-
-                /*
-                 * Creation of multiple VMs in a single process is not supported.
-                 * http://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html#JNI_CreateJavaVM
-                 */
-
-                BL_CHK_T(
-                    true,
-                    g_javaVMCreated,
-                    JavaException(),
-                    "JavaVM has already been created"
-                    );
-
-                const auto procAddress = os::getProcAddress( m_jvmLibrary, "JNI_CreateJavaVM" );
-
-                const auto jniCreateJavaVM = reinterpret_cast< jint ( JNICALL* )( JavaVM**, void**, void *) >( procAddress );
-
-                JavaVMInitArgs vmArgs = {};
-                vmArgs.version = JNI_VERSION_1_8;
-                vmArgs.nOptions = 0;
-                vmArgs.options = nullptr;
-                vmArgs.ignoreUnrecognized = JNI_FALSE;
+                BL_NOEXCEPT_BEGIN()
 
                 BL_LOG(
                     Logging::debug(),
                     BL_MSG()
-                        << "Creating JVM"
+                        << "Destroying JVM"
                     );
 
-                JNIEnv* jniEnv;
-                JavaVM* javaVM;
-
-                jint jniErrorCode = jniCreateJavaVM(
-                    &javaVM,
-                    ( void** )&jniEnv,
-                    &vmArgs
-                    );
+                const jint jniErrorCode = m_javaVM -> DestroyJavaVM();
 
                 BL_CHK_T(
                     false,
                     jniErrorCode == JNI_OK,
                     JavaException(),
                     BL_MSG()
-                        << "Failed to create JVM. ErrorCode "
+                        << "Failed to destroy JVM. ErrorCode "
                         << jniErrorCode
                         << " ["
                         << jniErrorMessage( jniErrorCode )
                         << "]"
                     );
 
-                m_javaVM = javaVM;
-                g_javaVMCreated = true;
+                BL_NOEXCEPT_END()
+            }
+
+        public:
+
+            static void setConfig( SAA_in JavaVirtualMachineConfig&& jvmConfig )
+            {
+                g_jvmConfig = BL_PARAM_FWD( jvmConfig );
+            }
+
+            static const JavaVirtualMachineConfig& getConfig() NOEXCEPT
```

Comment:
style: since the return type is long it is better to use the trailing type style